components/Login.js

```diff
@@ -0,0 +1,42 @@
+import { useSession, signIn, signOut } from "next-auth/react";
+import styled from "styled-components";
+
+const StyledLoginButton = styled.button`
+  background: var(--button-background);
+  color: var(--contrast-text);
+  border-style: none;
+  border-radius: 6px;
+`;
+
+const StyledLogoutButton = styled.button`
+  background: var(--button-background);
+  color: var(--contrast-text);
+  border-style: none;
+  border-radius: 6px;
+`;
+
+const StyledP = styled.p`
+  font-size: 0.8rem;
+  position: absolute;
+  top: 0;
+  right: 0;
+  margin-right: 0.8rem;
+`;
+
+export default function Login() {
+  const { data: session } = useSession();
+
+  if (session) {
+    return (
+      <>
+        <StyledP>Signed in as {session.user.name}</StyledP>
+        <StyledLogoutButton onClick={() => signOut()}>
+          Sign out
+        </StyledLogoutButton>
+      </>
+    );
+  }
+  return (
+    <StyledLoginButton onClick={() => signIn()}>Sign in</StyledLoginButton>
+  );
+}
```

Comment:
```suggestion
   return (
    <>
      {session ? (
        <>
          <StyledParagraph>Signed in as {session.user.name}</StyledParagraph>
          <StyledLogoutButton onClick={() => signOut()}>
            Sign out
          </StyledLogoutButton>
        </>
      ) : (
        <StyledLoginButton onClick={() => signIn()}>Sign in</StyledLoginButton>
      )}
    </>
  );
}
```
In React components, it's best practice to have only one return statement, as it helps maintain clarity and adheres to the principle of single responsibility, making the code easier to understand and maintain.